www/authmemcookie.php

```diff
@@ -91,12 +91,25 @@
 	/* Store the authentication data in the memcache server. */
 
 	$data = '';
+	$amcConfig = SimpleSAML_Configuration::getConfig('authmemcookie.php');
+	$format = strtolower($amcConfig->getBoolean('format', 'implode'));
+	assert('in_array($format, array("implode", "multi", "json"))');
 	foreach($authData as $n => $v) {
-		if(is_array($v)) {
-			$v = implode(':', $v);
+		if(!is_array($v)) {
+			$v = array((string)$v);
+		}
+		if ($format === 'multi') {
+			$v = array(implode(':', $v));
 		}
 
-		$data .= $n . '=' . $v . "\r\n";
+		if (in_array($format, array('implode', 'multi'))) {
```

Comment:
Since there's only a couple of options that we are comparing to, I suggest making the condition more readable (and slightly more efficient):

```php
if ($format === 'implode' || $format === 'multi') {
```